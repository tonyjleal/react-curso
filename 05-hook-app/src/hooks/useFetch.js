import React, { useEffect, useState } from 'react'

export const useFetch = (url) => {

    const [state, setState] = useState({ data:null, loading: true, error: null });
    
    useEffect( () => {

        setState({data:null, loading: true, error: null})

        fetch(url)
            .then( res => res.json() )
            .then( data => {
                setState({
                    data,
                    loading: false,
                    error: null,
                });
            });
    },[url]);

    return state;

    //const `https://www.breakingbadapi.com/api/quotes/${id}`


}
