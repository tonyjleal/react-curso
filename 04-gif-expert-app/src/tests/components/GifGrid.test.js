import React from "react";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    const category = 'One Punch';

    test('Debe mostrar el componente correctamente', () => {
        
        useFetchGifs.mockReturnValue({ 
            data: [], 
            loading: true,
        });
        
        const wrapper = shallow(<GifGrid category="" />);
        expect(wrapper).toMatchSnapshot();
        
    });
    
    test('debe de mostrar items cuando se cargan imÃ¡genes con useFetchGifs', () => {

        const gifs = [{
            id: 'id',
            url: 'https://localhost/sample',
            title: 'title'
        }]

        useFetchGifs.mockReturnValue({ 
            data: gifs, 
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();

    });
    
    


});
