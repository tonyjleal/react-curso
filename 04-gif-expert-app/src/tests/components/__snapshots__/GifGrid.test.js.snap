// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  />
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="id"
      key="id"
      title="title"
      url="https://localhost/sample"
    />
  </div>
</Fragment>
`;
